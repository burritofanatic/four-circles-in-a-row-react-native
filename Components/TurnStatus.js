import React, {Component} from 'react';
import {View, Text, StyleSheet} from 'react-native';
import * as Animatable from 'react-native-animatable';

class TurnStatus extends Component {

  render() {
    let text;
    if (this.props.gameOver) {
      if (this.props.winner == 3) {
        text = 'Tied!';
      } else if (this.props.userIsWinner) {
        text = 'You Win!';
      } else {
        text = 'You Lose!';
      }
    } else {
      if (this.props.isTurn) {
        text = 'Your Turn!';
      } else {
        text = 'Their Turn!';
      }
    }

    return (
        <Animatable.Text style={styles.alertText}
                      animation='pulse' easing='ease-out'
                      iterationCount='infinite'>{text}
                      </Animatable.Text>
    );
  }
}

const styles = StyleSheet.create({
  alertText: {
    color: 'white',
    fontSize: 40
  }
});

export default TurnStatus;
