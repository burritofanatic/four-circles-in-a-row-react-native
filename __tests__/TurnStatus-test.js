import React from 'react';
import TurnStatus from '../Components/TurnStatus';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
        .create(<TurnStatus winner={3} gameOver={true}/>)
        .toJSON();
  expect(tree).toMatchSnapshot();
})
