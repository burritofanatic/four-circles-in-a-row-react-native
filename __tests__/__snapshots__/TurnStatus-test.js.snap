// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "white",
      "fontSize": 40,
      "transform": Array [
        Object {
          "scale": 1,
        },
      ],
    }
  }
>
  Tied!
</Text>
`;
